### PR TITLE
PageHeader: Make default heading level `h2` 

### PR DESCRIPTION
**Pull request by broccolinisoup**
This PR changes the `PageHeder.Title`'s default heading level to `h2` as discussed [here](https://github.com/github/primer/issues/1571#issuecomment-1434069132)

Also added a feature story and docs to demonstrate the cases where components are used as a title instead of a typical text title and included a visually hidden title to set an example. 

Closes https://github.com/github/primer/issues/1571

### Screenshots

No visual changes.

### Merge checklist
- [x] Added/updated documentation
- [x] Tested in Chrome

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: src/PageHeader/features.stories.tsx [hunk 1]
```diff
@@ -65,6 +65,24 @@ export const WithLeadingAndTrailingVisuals = () => (
...
+      <PageHeader.TitleArea>
+        <Breadcrumbs>
+          <Breadcrumbs.Item href="#">...</Breadcrumbs.Item>
+          <Breadcrumbs.Item href="#">primer</Breadcrumbs.Item>
+          <Breadcrumbs.Item href="#">react</Breadcrumbs.Item>
+          <Breadcrumbs.Item href="#">src</Breadcrumbs.Item>
+          <Breadcrumbs.Item href="#">PageHeader</Breadcrumbs.Item>
+          <Breadcrumbs.Item href="#">PageHeader.tsx</Breadcrumbs.Item>
+        </Breadcrumbs>
+        <h2 className="visually-hidden">Visually Hidden Title</h2>
```
Comment (TylerJDev) [inline src/PageHeader/features.stories.tsx:80]: I noticed that in the [preview for this example](https://primer-5c67b4ddbf-13348165.drafts.github.io/storybook/?path=/story/drafts-components-pageheader-features--with-component-as-a-title), the heading isn't actually hidden. Is this expected?

**Comment by TylerJDev** [comment]
Slightly off-topic, but [within the docs under the `as` prop](https://primer-5c67b4ddbf-13348165.drafts.github.io/drafts/PageHeader#pageheader), we state "`header`" as the default value. I wonder if we'd want 
"`div`" here?

**Review by TylerJDev — APPROVED** [review]
Overall looks good! I left some comments that might require some discussion, but other than that I think it looks good. Thanks for this too! :grin:

File: src/PageHeader/features.stories.tsx (hunk 1 above)
Comment (broccolinisoup) [reply src/PageHeader/features.stories.tsx:80]: 😬 No it is NOT EXPECTED 😅 I'll fix it

**Comment by broccolinisoup** [comment]
> within the docs under the `as` prop

Oh good catch! On the code it is `div` but apparently I got it wrong on the docs. I'll update it.

**Comment by broccolinisoup** [comment]
@TylerJDev I pushed a commit to make the visually hidden titles visually hidden for real 😅 and fix the docs about the default `as`.

You can see [in the storybook ](https://github.com/primer/react/pull/2905/files#diff-658515da38d13062047b299826a280a39be6d31c08e5d8fed4f79f357e1d5e1cR81)I used `VisuallyHidden` component whereas I added [styles via sx prop](https://github.com/primer/react/pull/2905/files#diff-6a41020167cf9d1f245b8f5d8bb73c6492b65ac577e3f576e0457e7751c3dc19R85) on the [primer.style docs](https://primer-5c67b4ddbf-13348165.drafts.github.io/drafts/PageHeader#component-as-title). My motivation behind this was about the docs practicality. Primer.style docs examples should just work when people copy and paste the code snippets provided and given the fact that the `VisuallyHidden` component is an internal util component, I manually added its styles via sx prop. Let me know if you have any concern 🙌🏻

**Comment by TylerJDev** [comment]
@broccolinisoup - Looks great 🤩. Thank you again!

**Review by joshblack — APPROVED** [review]

**Pull request merged**